inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java

```diff
@@ -138,11 +144,30 @@ public class AgentServiceImpl implements AgentService {
      */
     @PostConstruct
     private void startHeartbeatTask() {
-        if (enabled) {
-            UpdateTaskRunnable taskRunnable = new UpdateTaskRunnable();
-            this.executorService.execute(taskRunnable);
+        if (updateTaskTimeoutEnabled) {
+            ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+            executor.scheduleWithFixedDelay(() -> {
+                try {
+                    sourceMapper.updateStatusToTimeout(beforeSeconds);
+                    LOGGER.info("update task status successfully");
+                } catch (Throwable t) {
+                    LOGGER.error("update task status error", t);
+                }
+            }, 0, beforeSeconds, TimeUnit.SECONDS);
```

Comment:
1. Suggest add a new interval params, not reuse the `beforeSeconds`.
2. Please add a name for the thread pool, such as: 
    ```java
    ThreadFactory factory = new ThreadFactoryBuilder()
            .setNameFormat("scheduled-source-timeout-%d")
            .setDaemon(true)
            .build();
    ScheduledExecutorService executor  = Executors.newSingleThreadScheduledExecutor(factory);
    ```

Reply:
done